Source/SwiftLintCore/Models/Baseline.swift

```diff
@@ -0,0 +1,169 @@
+import Foundation
+
+private typealias GroupedViolations = [String: [BaselineViolation]]
+
+private struct BaselineViolation: Equatable, Codable, Hashable {
+    let violation: StyleViolation
+    let text: String
+    var key: String { text + violation.reason + violation.severity.rawValue }
+
+    init(violation: StyleViolation, text: String) {
+        let location = violation.location
+        self.violation = violation.with(location: Location(
+            // Within the baseline, we use relative paths, so that
+            // comparisons are independent of the absolute path
+            file: location.relativeFile,
+            line: location.line,
+            character: location.character)
+        )
+        self.text = text
+    }
+}
+
+/// A set of violations that can be used to filter newly detected violations.
+public struct Baseline: Equatable {
+    private let baselineViolations: GroupedViolations
+
+    /// The stored violations.
+    public var violations: [StyleViolation] {
+        baselineViolations.keys.sorted().flatMap({ baselineViolations[$0]! }).violationsWithAbsolutePaths
+    }
+
+    /// Creates a `Baseline` from a saved file.
+    ///
+    /// - parameter fromPath: The path to read from.
+    public init(fromPath path: String) throws {
+        let data = try Data(contentsOf: URL(fileURLWithPath: path))
+        baselineViolations = try JSONDecoder().decode(GroupedViolations.self, from: data)
+    }
+
+    init(violations: [StyleViolation]) {
+        baselineViolations = violations.baselineViolations.groupedByFile()
+    }
+
+    /// Writes a `Baseline` to disk in JSON format.
+    ///
+    /// - parameter violations: The violations to save.
+    /// - parameter toPath: The path to write to.
+    public static func write(_ violations: [StyleViolation], toPath path: String) throws {
```

Comment:
I'd expect the call-side to be like `Baseline(violations:).write(toPath:)`, in other words, the `static` can be avoided.